sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -78,6 +79,26 @@ final class DataFrameWriter private[sql](df: DataFrame) {
   }
 
   /**
+   * Set the trigger period for the stream query.
+   *
+   * @since 2.0.0
+   */
+  def trigger(period: Duration): DataFrameWriter = {
+    this.extraOptions += ("period" -> period.toMillis.toString)
```

Comment:
Since this name is exposed for the user to accidentally modify (through option() ) we probably should make this more specific. Maybe "triggerInterval"